### PR TITLE
Add GlobalTracer and ActiveSpan

### LINKED ISSUE #96: Get Span from ctx
Sometime I need spanFromContext

such as [go-redis-hook](https://github.com/go-redis/redis/blob/55db07d5d106172e712d815ad3185084a1c61f5c/redis.go#L25)

```
type Hook interface {
	BeforeProcess(ctx context.Context, cmd Cmder) (context.Context, error)
	AfterProcess(ctx context.Context, cmd Cmder) error
}
```

only ctx can be propagation, I cant do span.End further

Maybe there is a easy way to solve, set ctxKeyInstance to be public. Then I can use this key to get activeSpan from ctx and do (.Span) to get a Span

### PR DESCRIPTION
**Pull request by arugal**
### GlobalTracer
```go
tracer, _ := NewTracer("service")

// registers `tracer` as the global Tracer.
SetGlobalTracer(tracer)

// returns the registered global Tracer
tracer1 := GetGlobalTracer()
```

### ActiveSpan
```go
tracer, _ := NewTracer("service")
_, ctx, err := tracer.CreateLocalSpan(context.Background())
if err != nil {
  t.Error(err)
}

activeSpan := ActiveSpan(ctx)
```
resolve #96

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,40 +17,43 @@ The API of this project is still evolving. The use of vendoring tool is recommen
...
+to export to OAP server, which is listening on `oap-skywalking` port `11800`, and all the spans from this program will
+be associated with a service name `example`.
+`reporter.GRPCReporter` can also adjust the behavior through `reporter.GRPCReporterOption`
+, [view all](docs/GRPC-Reporter-Option.md).
+
 ```go
 r, err := reporter.NewGRPCReporter("oap-skywalking:11800")
 if err != nil {
-    log.Fatalf("new reporter error %v \n", err)
+log.Fatalf("new reporter error %v \n", err)
```
Comment (wu-sheng) [inline README.md:34]: Mis-formated?

File: README.md [hunk 2]
```diff
@@ -16,40 +17,43 @@ The API of this project is still evolving. The use of vendoring tool is recommen
...
 }
 defer r.Close()
 tracer, err := go2sky.NewTracer("example", go2sky.WithReporter(r))
-// create with sampler
-// tracer, err := go2sky.NewTracer("example", go2sky.WithReporter(r), go2sky.WithSampler(0.5))
```
Comment (wu-sheng) [inline README.md:36]: Why remove this?

File: README.md [hunk 3]
```diff
@@ -160,37 +188,38 @@ exitSpan, err := tracer.CreateExitSpan(entryCtx, ...)
...
+service.
 
 `Entry` and `Exit` spans make sense to OAP analysis which generates topology map and service metrics.
 
 ```go
 //Extract context from HTTP request header `sw8`
-span, ctx, err := tracer.CreateEntrySpan(r.Context(), "/api/login", func(key string) (string, error) {
-		return r.Header.Get(key), nil
+span, ctx, err := tracer.CreateEntrySpan(r.Context(), "/api/login", func (key string) (string, error) {
+return r.Header.Get(key), nil
```
Comment (wu-sheng) [inline README.md:199]: This seems mis-format too.

File: README.md [hunk 4]
```diff
@@ -160,37 +188,38 @@ exitSpan, err := tracer.CreateExitSpan(entryCtx, ...)
...
 // Some operation
 ...
 
 // Inject context into HTTP request header `sw8`
-span, err := tracer.CreateExitSpan(req.Context(), "/service/validate", "tomcat-service:8080", func(key, value string) error {
-		req.Header.Set(key, value)
-		return nil
+span, err := tracer.CreateExitSpan(req.Context(), "/service/validate", "tomcat-service:8080", func (key, value string) error {
+req.Header.Set(key, value)
+return nil
```
Comment (wu-sheng) [inline README.md:208]: Same here.

File: README.md (hunk 2 above)
Comment (arugal) [reply README.md:36]: It has been described here

https://github.com/SkyAPM/go2sky/blob/0f9874e2d30cb5e4b3e4721c0830c52eb8c0f97f/README.md?plain=1#L39-L43

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**